Update the doc of adding AAD B2C (#517)
MicrosoftDocs/azure-dev-docs: articles/java/spring-framework/configure-spring-boot-starter-java-app-with-azure-active-directory-b2c-oidc.md

   ![Screenshot of the Azure portal showing the Azure AD B2C App registrations screen](media/configure-spring-boot-starter-java-app-with-azure-active-directory-b2c-oidc/b2c1-n.png)

> 2. In the **Name** field, enter your app's name, select **Register**.

Reviewer: ```suggestion
2. In the **Name** field, enter your app's name, then select **Register**.
```

   ![configure a new app registration](media/configure-spring-boot-starter-java-app-with-azure-active-directory-b2c-oidc/b2c4-n.png)

> 3. Back on the **Manage** pane , select **App registrations** , then select the application name you created.

Reviewer: ```suggestion
3. Back on the **Manage** pane, select **App registrations**, then select the application name you created.
```

   ![Update application](media/configure-spring-boot-starter-java-app-with-azure-active-directory-b2c-oidc/b2c5-n.png)

> 4. Select **Authentication** then click **Add a platform** then select **Web**. Set the **Redirect URIs** to `http://localhost:8080/login/oauth2/code/`. Click **Configure**.

Reviewer: ```suggestion
4. Select **Authentication**, then **Add a platform** then **Web**. Set the **Redirect URIs** to `http://localhost:8080/login/oauth2/code/`, then select **Configure**.
```

   
  ![Add authentication](media/configure-spring-boot-starter-java-app-with-azure-active-directory-b2c-oidc/b2c7-n.png)
  ![Add authentication](media/configure-spring-boot-starter-java-app-with-azure-active-directory-b2c-oidc/b2c8-n.png)

> ### Add app secrets for your app.    

Reviewer: ```suggestion
### Add app secrets for your app
```

>    Select **Certificates & secrets** then click **New client secrets**, then input your secret description and click **ADD**. After you create the secret, do remember to copy the secret value for use later in this article. There is a copy icon next to the secret value, click it to copy the value.

Reviewer: ```suggestion
   Select **Certificates & secrets**, then **New client secrets**. Input your secret description and then select **ADD**. After you create the secret, select the copy icon next to the secret value to copy the value for use later in this article.
```

    
   ![Update app key](media/configure-spring-boot-starter-java-app-with-azure-active-directory-b2c-oidc/b2c9-n.png)
   ![Update app key](media/configure-spring-boot-starter-java-app-with-azure-active-directory-b2c-oidc/b2c10-n.png)

    > [!NOTE]
    > If you leave the **Keys** section and come back, you will not be able to see the key value. In that case, you must create another key and copy it for future use.
    > Occasionally, the generated key may contain characters that are problematic for inclusion in the *application.yml* file, such as backslash or backtick. In that case, discard that key and generate another one.

    ![Create the secret](media/configure-spring-boot-starter-java-app-with-azure-active-directory-b2c-oidc/b2c3-n.png)

> ### Add user flow.

Reviewer: ```suggestion
### Add user flow
```

1. Navigate to your tenant main page. In the **Policies** section of the left pane, select **User flows**, then select **New user flow**.

> 2. You will now leave this tutorial, execute another tutorial, and come back to this tutorial when you are done. Here are some things to keep in mind when you go to the other tutorial.

Reviewer: ```suggestion
2. You will now leave this tutorial, execute another tutorial, and come back to this tutorial when you're done. Here are some things to keep in mind when you go to the other tutorial.
```

    ```xml
    <dependency>
        <groupId>com.azure.spring</groupId>
        <artifactId>azure-spring-boot-starter-active-directory-b2c</artifactId>
        <version>See Below</version>
    </dependency>
    <dependency>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-thymeleaf</artifactId>
    </dependency>
    <dependency>
        <groupId>org.thymeleaf.extras</groupId>
        <artifactId>thymeleaf-extras-springsecurity5</artifactId>
    </dependency>
    ```

>     For the `azure-spring-boot-starter-active-directory-b2c`, use the latest version available. You may be able to use [mvnrepository.com](https://mvnrepository.com/ artifact/com.microsoft.azure/azure-active-directory-spring-boot-starter) to look this up. As of this updating the latest version is `3.4.0`.

Reviewer: ```suggestion
    For the `azure-spring-boot-starter-active-directory-b2c`, use the latest version available. You may be able to use [mvnrepository.com](https://mvnrepository.com/ artifact/com.microsoft.azure/azure-active-directory-spring-boot-starter) to look this up. At the time of this update, the latest version is `3.4.0`.
```

    azure:
      activedirectory:
        b2c:
          base-uri: https://<your-tenant-initial-domain-name>.b2clogin.com/<your-tenant-initial-domain-name>.onmicrosoft.com/

>           client-id: <your-application-id> 

Reviewer: ```suggestion
          client-id: <your-application-ID>
```

>           client-secret: '<secretValue>'

Reviewer: ```suggestion
          client-secret: '<secret-value>'
```

          login-flow: sign-up-or-sign-in

>           logout-success-url: <your-logout-success-url> 

Reviewer: ```suggestion
          logout-success-url: <your-logout-success-URL>
```

          user-flows:
            sign-up-or-sign-in: <your-sign-up-or-sign-in-user-flow-name> 
            profile-edit: <your-profile-edit-user-flow-name> 
            password-reset: <your-password-reset-user-flow-name> 
          user-name-attribute-name: <your-user-name-attribute-name> 
    ```

>     Notice that the `client-secret` value is enclosed in single quotes. This is necessary because the value of `<secretValue>` will almost certainly contain some characters that require being inside single quotes when present in YAML.

Reviewer: ```suggestion
    Notice that the `client-secret` value is enclosed in single quotes. This is necessary because the value of `<secret-value>` will almost certainly contain some characters that require being inside single quotes when present in YAML.
```

    ![Successful login](media/configure-spring-boot-starter-java-app-with-azure-active-directory-b2c-oidc/lo3-n.png)

## Troubleshooting
- `Missing attribute 'name' in attributes `

>   ```

Reviewer: ```suggestion
  ```output
```
